habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -1162,81 +1178,121 @@ def get_runtime_perf_stats(self) -> Dict[str, float]:
 
         return stats_dict
 
-    def set_receptacle_in_scene(self, ep_info):
+    def set_receptacle_in_scene(self, ep_info, receptacle_name=None):
         rom = self.get_rigid_object_manager()
         otm = self.get_object_template_manager()
+        hssd = 'dataset' in ep_info.info and ep_info.info['dataset'] == 'hssd'
+        mp3d = 'dataset' in ep_info.info and ep_info.info['dataset'] == 'mp3d'
+        if hssd:
+            for object_path in ep_info.additional_obj_config_paths:
+                otm.load_configs(osp.abspath(object_path))
         assert ep_info.target_receptacles is not None and ep_info.goal_receptacles is not None
         receptacles, target_receptacles, goal_receptacles = [], [], []
         info = {}
+        recep_settled = []
         for i, (tar_recep_handle, tar_tranform, tar_translation) in enumerate(ep_info.target_receptacles):
-            tar_recep_name = tar_recep_handle
-            if '_:' in tar_recep_name:
-                tar_recep_name = tar_recep_handle[:-6]
-            tar_template = otm.get_templates_by_handle_substring(
-                tar_recep_name
-            )
-            if not tar_template:
-                raise ValueError(
-                    f"Template not found for object with handle {tar_recep_handle}"
+            if (
+                (
+                    receptacle_name and
+                    receptacle_name in tar_recep_handle and
+                    tar_recep_handle not in recep_settled
                 )
-            tar_path = list(tar_template.keys())[0]
-            target_receptacle = rom.add_object_by_template_handle(
-                tar_path
-            )
-            if self._kinematic_mode:
-                target_receptacle.motion_type = habitat_sim.physics.MotionType.KINEMATIC
-                target_receptacle.collidable = False
+                or receptacle_name is None
+            ):
+                recep_settled.append(tar_recep_handle)
+                tar_recep_name = tar_recep_handle
+                if '_:' in tar_recep_name:
+                    tar_recep_name = tar_recep_handle[:-6]
+                tar_template = otm.get_templates_by_handle_substring(
+                    tar_recep_name
+                )
+                if not tar_template:
+                    raise ValueError(
+                        f"Template not found for object with handle {tar_recep_handle}"
+                    )
+                tar_path = list(tar_template.keys())[0]
+                with open(tar_path) as tar_file:
+                    tar_object_config = json.load(tar_file)
+                user_defined = tar_object_config['user_defined']
+                if 'scale' in user_defined:
+                    tar_scale = user_defined['scale']
+                else:
+                    tar_scale = user_defined[list(user_defined.keys())[0]]['scale']
```

Comment:
Do we also randomize the target object scale when inserting new objects into the scene? Why do we need this?

Reply:
Since the scales in replica dataset and hssd dataset are different, to set the wall cabinet (which is a receptacle in replica dataset) properly, we need adjust the scale of it. 
For the question about the random scale, the simulator doesn't randomly init the scale either. It just set the scale as (1,1,1).